lib/administrate/base_dashboard.rb

```diff
@@ -7,6 +7,7 @@
 require "administrate/field/has_one"
 require "administrate/field/number"
 require "administrate/field/polymorphic"
+require "administrate/field/rich_text"
```

Comment:
As an alternative to eager-loading the file, have you considered utilizing Rails' [Engine load hooks](https://guides.rubyonrails.org/engines.html#available-load-hooks)

If we relied on the host application depending on the `action_text` gem and loading `action_text/engine`, then we could defer loading the `Administrate::Field::RichText`.

I've tried this out locally with the following changes:

```diff
diff --git a/Gemfile.lock b/Gemfile.lock
index c740df4..a890f52 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -3,7 +3,6 @@ PATH
   specs:
     administrate (0.19.0)
       actionpack (>= 5.0)
-      actiontext (>= 6.0)
       actionview (>= 5.0)
       activerecord (>= 5.0)
       jquery-rails (>= 4.0)
diff --git a/administrate.gemspec b/administrate.gemspec
index b6dbad7..dc704d1 100644
--- a/administrate.gemspec
+++ b/administrate.gemspec
@@ -16,7 +16,6 @@ Gem::Specification.new do |s|
   s.add_dependency "actionpack", ">= 5.0"
   s.add_dependency "actionview", ">= 5.0"
   s.add_dependency "activerecord", ">= 5.0"
-  s.add_dependency "actiontext", ">= 6.0"
 
   s.add_dependency "jquery-rails", ">= 4.0"
   s.add_dependency "kaminari", ">= 1.0"
diff --git a/lib/administrate/engine.rb b/lib/administrate/engine.rb
index 3e6dc7b..00158a2 100644
--- a/lib/administrate/engine.rb
+++ b/lib/administrate/engine.rb
@@ -1,4 +1,3 @@
-require "action_text/engine"
 require "jquery-rails"
 require "kaminari"
 require "sassc-rails"
@@ -30,6 +29,12 @@ module Administrate
       ]
     end
 
+    initializer "administrate.action_text" do
+      ActiveSupport.on_load :action_text_rich_text do
+        require "administrate/field/rich_text"
+      end
+    end
+
     def self.add_javascript(script)
       @@javascripts << script
     end
diff --git a/spec/example_app/app/controllers/application_controller.rb b/spec/example_app/app/controllers/application_controller.rb
index 2fc8543..c17b696 100644
--- a/spec/example_app/app/controllers/application_controller.rb
+++ b/spec/example_app/app/controllers/application_controller.rb
@@ -1,5 +1,3 @@
-require "action_text"
-
 class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
diff --git a/spec/example_app/config/application.rb b/spec/example_app/config/application.rb
index 82bbe4f..ebb45c7 100644
--- a/spec/example_app/config/application.rb
+++ b/spec/example_app/config/application.rb
@@ -5,6 +5,7 @@ require "active_record/railtie"
 require "action_controller/railtie"
 require "action_view/railtie"
 require "sprockets/railtie"
+require "action_text/engine"
 
 # Require the gems listed in Gemfile, including any gems
 # you've limited to :test, :development, or :production.
```

Reply:
@seanpdoyle I tried to implement this, but then I am back to the `undefined method `rails_direct_uploads_url'` error.

Reply:
But not when I run the specs individually, hmmm.

Reply:
Still getting these errors when running the test suite as a whole, but not when run individually, even after all of our additions :(

Reply:
Oh, forgot about [this comment](https://github.com/thoughtbot/administrate/pull/2411#issuecomment-1693583352)

Reply:
Fixed in 7304e0b47c3fa6e97c63bc685892a6d16c23c358

Reply:
@seanpdoyle ultimately I can't get it to work when I remove this `require "administrate/field/rich_text"` (with the changes in `lib/administrate/engine.rb`). Any other ideas, or should I just remove the changes in `lib/administrate/engine.rb`?